src/charm.py

```diff
@@ -453,7 +495,8 @@ def mongos_config(self) -> MongosConfiguration:
             username=self.get_secret(APP_SCOPE, Config.Secrets.USERNAME),
             password=self.get_secret(APP_SCOPE, Config.Secrets.PASSWORD),
             hosts=hosts,
```

Comment:
```suggestion
            hosts=set(hosts),
```

Reply:
The constructor of `MongosConfiguration` takes a `hosts: set` as an arg, not a `list`. 
The variable `hosts` passed here is of type `list`. Instead of creating a new set - you can declare it as a set instead of a list a few lines previously.

Reply:
I do agree that we should build a set from the beginning and then keep it as is to respect typing (and here it's good to use a set to ensure we have each host only once)

Reply:
will update the call itself on line 489